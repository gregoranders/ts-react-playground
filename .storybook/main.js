const path = require('path');

module.exports = {
  typescript: {
    check: false,
    checkOptions: {},
    reactDocgen: 'react-docgen-typescript',
    reactDocgenTypescriptOptions: {
      shouldExtractLiteralValuesFromEnum: true,
      propFilter: (prop) =>
        prop.parent ? !/node_modules/.test(prop.parent.fileName) : true,
    },
  },

  webpackFinal: async (config) => {
    config.module.rules.push({
      test: /\.(ts|tsx)$/,
      use: [
        {
          loader: require.resolve('ts-loader'),
        },
        // Optional
        {
          loader: require.resolve('react-docgen-typescript-loader'),
        },
      ],
    });

    config.module.rules.push({
      test: /\.scss$/,
      use: ['style-loader', 'css-loader', 'sass-loader'],
      include: path.resolve(__dirname, '../'),
    });

    config.resolve.alias = {
      '@models': path.resolve(__dirname, '..', 'src/script/models'),
      '@states': path.resolve(__dirname, '..', 'src/script/states'),
      '@atoms': path.resolve(__dirname, '..', 'src/script/components/atoms'),
      '@organisms': path.resolve(
        __dirname,
        '..',
        'src/script/components/organisms',
      ),
      '@molecules': path.resolve(
        __dirname,
        '..',
        'src/script/components/molecules',
      ),
      '@pages': path.resolve(__dirname, '..', 'src/script/components/pages'),
      '@components': path.resolve(__dirname, '..', 'src/script/components'),
      '@app': path.resolve(__dirname, '..', 'src/script'),
      './fonts': 'node_modules/@fortawesome/fontawesome-free/webfonts',
    };

    config.performance = {
      hints: false,
      maxEntrypointSize: 200000000,
      maxAssetSize: 200000000,
    };

    config.resolve.extensions.push('.ts', '.tsx', '.scss');

    return config;
  },

  stories: ['../src/script/**/*.stories.@(tsx|mdx)'],

  addons: [
    '@storybook/addon-actions',
    '@storybook/addon-links',
    '@storybook/addon-knobs',
    '@storybook/addon-jest',
    '@storybook/addon-backgrounds',
    '@storybook/addon-docs' /* '@storybook/addon-controls', */,
    '@storybook/addon-viewport',
    '@storybook/addon-a11y',
  ],
};
