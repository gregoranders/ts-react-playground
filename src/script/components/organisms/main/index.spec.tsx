import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import * as TestSubject from '@organisms/main';

describe(`${TestSubject.Main.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  describe('snapshot', () => {
    it('string', (): void => {
      const snapshot = create(<TestComponent>test</TestComponent>);
      snapshot.toJSON();
      expect(snapshot).toMatchSnapshot();
    });

    it('element', (): void => {
      const snapshot = create(
        <TestComponent>
          <span>test</span>
        </TestComponent>,
      );
      snapshot.toJSON();
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('mount', () => {
    it('string', () => {
      const testSubject = mount(<TestComponent>test</TestComponent>);
      expect(testSubject.text()).toBe('test');
    });

    it('string', () => {
      const testSubject = mount(
        <TestComponent>
          <span>test</span>
        </TestComponent>,
      );
      expect(testSubject.text()).toBe('test');
    });
  });
});
