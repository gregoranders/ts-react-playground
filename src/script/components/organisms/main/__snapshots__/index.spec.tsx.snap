// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot element 1`] = `
<main
  className="w3-margin"
>
  <span>
    test
  </span>
</main>
`;

exports[`Main snapshot string 1`] = `
<main
  className="w3-margin"
>
  test
</main>
`;
